connectivity/check/metrics.go

```diff
@@ -113,9 +113,24 @@ func parseMetrics(reader io.Reader) (promMetricsFamily, error) {
 
 // metricsIncrease verifies for all the metrics that the values increased.
 func metricsIncrease(mf1, mf2 *dto.MetricFamily) error {
-	metrics1 := mf1.GetMetric()
 	metrics2 := mf2.GetMetric()
 
+	// Absent initial metric means it's zero, check that after metric is non-zero.
+	if mf1 == nil && mf2 != nil {
+		for _, m2 := range metrics2 {
+			if m2.GetCounter() == nil {
+				return fmt.Errorf("metric %s is not a Counter: %v", mf2.GetName(), mf2)
+			}
+
+			if val := m2.GetCounter().GetValue(); val == 0.0 {
+				return fmt.Errorf("metric %s did not increase as expected, value 1: %f and value 2: %f", mf2.GetName(), 0.0, val)
+			}
+		}
+		return nil
+	}
+
+	metrics1 := mf1.GetMetric()
```

Comment:
`mf1` can be nil right?

Reply:
Yes but if `mf2==nil`, we should have catch where the method is called meaning if mf1 is nil we should have failed before. 
I know that's a bit tricky...

Reply:
Thanks! Moved at the top `mf1.GetMetric()`